src/test/java/echoserver/ServiceTest.java

```diff
@@ -50,17 +52,17 @@ public void setupService() {
 
     @Test
     public void whenTheServerStarts_clientInputIsLogged() {
-        serverThread = new Service(client, echoLogger);
-        serverThread.run();
+        serverThread = new Service(echoLogger);
+        serverThread.runWith(client);
```

Comment:
Some of the naming is starting to get confusing for me. This `client` is a `ConnectionDataStream`. Why not call it a `connectionDataStream`? Or is it a `clientConnection`, in which case, why not call it a `clientConnection`?

Reply:
Something I have been wondering is would renaming `ClientConnection` to `Connection` be worthwhile? Does *client* add much? `ConnectionDataStream` also feels pretty specific.

Perhaps `ConnectionDataStream` -> `Connection`
And `ClientConnection` -> `SocketConnection`?